src/Sparrow.Server/Tensors/Functions.cs

```diff
@@ -162,6 +162,21 @@ public static TResult CosineDistance<T, TResult>(ReadOnlySpan<T> a, TResult aMag
         {
             return TResult.One - CosineSimilarity(a, aMagnitude, b, bMagnitude);
         }
+        
+        [MethodImpl(MethodImplOptions.AggressiveOptimization | MethodImplOptions.AggressiveInlining)]
+        internal static TResult CosineSimilarityNormalize<T, TResult>(T ab, T a2, T b2)
+            where T : unmanaged, IRootFunctions<T>, INumber<T>
+            where TResult : unmanaged, IFloatingPoint<TResult>, IRootFunctions<TResult>, INumber<TResult>
+        {
+            if (Sse2.IsSupported)
+                return Vectorized256.CosineSimilarityNormalizeSse<T, TResult>(ab, a2, b2);
+
+            if (AdvInstructionSet.Arm.IsSupported)
+                return Vectorized128.CosineSimilarityNormalizeNeon<T, TResult>(ab, a2, b2);
```

Comment:
`CosineSimilarityIntegersNeon` uses `Vector128`, which is supported on Raspberry Pi; however, it also uses `Dp.DotProduct`, which I believe is not supported. Does `Dp.DotProduct` have a scalar overload for this scenario?

Reply:
Those 2 are independent, and they may be used in the future with SVE which is independent.